contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -5064,6 +5050,26 @@ makeUseStatement(TSqlParser::Use_statementContext *ctx)
 	return (PLtsql_stmt *) result;
 }
 
+PLtsql_stmt *
+makeKillStatement(TSqlParser::Kill_statementContext *ctx)
+{
+	PLtsql_stmt_kill *result = (PLtsql_stmt_kill *) palloc0(sizeof(*result));
+
+	result->cmd_type = PLTSQL_STMT_KILL;
+	result->lineno = getLineNo(ctx);
+
+	// Only supporting numeric argument for the spid,
+	// other flavours of KILL are intercepted in the parser
+
+	std::string spidStr;
+	if (ctx->kill_process()) {
+		spidStr = ::getFullText(ctx->kill_process());
+		result->spid = psprintf("%s", &spidStr[0]);
```

Comment:
Not sure why you're doing it this way. A better way to assign it is:
`result->spid = = pstrdup(spidStr.c_str())`

Reply:
Also, why cant we store it as an integer datatype?

Reply:
Changed to result->spid = = pstrdup(spidStr.c_str()) as suggested.

Reply:
done